RIOT-OS/RIOT
# scapy_mqttsn: import MQTT-SN layer for scapy

### Contribution description
I provided [this also upstream][1] but it might take a while until there is a version of that one can install with `pip` etc (not to mention until the CI builders are sufficiently updated...). So for now I'd like to include this into our repo to provide some tests for our two MQTT-SN libraries, `emcute` and `asymcute`.

[1]: https://github.com/secdev/scapy/pull/2131




### Testing procedure
Clone my contribution to scapy:

```sh
cd ../
git clone --branch mqttsn_layer https://github.com/miri64/scapy
```

Make sure the python file to this repo is identical to the python file I contributed to scapy

```sh
diff -u scapy/scapy/contrib/mqttsn.py RIOT/dist/pythonlibs/scapy_mqttsn.py
```

(the output should be empty)

Run the [tests I provided to scapy](https://github.com/miri64/scapy/blob/mqttsn_layer/test/contrib/mqttsn.uts) for python 3 with scapy test suite:

```sh
cd scapy
./test/run_tests_py3 -P "load_contrib('mqttsn')" -t test/contrib/mqttsn.uts
```




### Issues/PRs references
None, but might help later on with #11811
